### ISSUE TITLE
`unicorn/no-array-callback-reference` does not report when optional chaining

### ISSUE THREAD

**Issue by czizzy** [sindresorhus/eslint-plugin-unicorn#2294]
this rule is not reported when optional chaining

`unicorn/no-array-callback-reference`

```js
function isString(arg) {
  return typeof arg === 'string';
}

const obj = { a: [], b: [] };

obj.a.filter(isString);
obj.b?.filter(isString);
```

only the first filter report an error.

**Comment by fisker** [comment]
Remove this line should fix this

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/231529aee793251f30416b93dc3d5f9bfa27ee47/rules/no-array-callback-reference.js#L238


PR welcome.
